src/client/components/LetterBox.tsx

```diff
@@ -11,14 +33,48 @@ export const LetterBox = ({
   guess?: string;
 }) => {
   const [displayedLetter, setDisplayedLetter] = useState("");
+  // @ts-ignore
+  const flipped = displayedLetter.length == true;
+  const { transform } = useSpring({
+    transform: `perspective(500px) rotateX(${flipped ? 180 : 0}deg)`,
+    config: { mass: 5, tension: 500, friction: 80 },
+  });
+
+  const transitions = useTransition(displayedLetter, {
+    from: hide,
+    enter: show,
+    leave: hide,
+    config: { mass: 5, tension: 500, friction: 80 },
+  });
 
   useEffect(() => {
     if (letter && guess && letter.toLowerCase() === guess.toLowerCase()) {
       setDisplayedLetter(letter);
     }
   }, [letter, guess]);
 
-  return <div data-testid="letter-box">{displayedLetter}</div>;
+  return (
+    <div style={boxStyle as React.CSSProperties}>
+      {transitions(({ opacity }, letter) => {
```

Comment:
Try to think of better name for `letter` to avoid confusion with `letter` from the outer scope